cmd/builder/Dockerfile

```diff
@@ -0,0 +1,12 @@
+FROM golang:alpine
```

Comment:
ocb requires a golang environment to function correctly; this seemed like a reasonable way to achieve that. Starting from scratch and installing go would be an alternative route that I'd be happy to explore, though.

Reply:
I also just submitted https://github.com/open-telemetry/opentelemetry-collector/issues/11382 to request removing "go binary" dependency for ocb; would allow for a slimmer image if possible.

Reply:
Oh, right, of course! The container image needs to include a Go compiler for the compilation phase to succeed. I suppose using `golang:alpine` is a good choice in this case. Let's just pin the Go version and Alpine version to prevent any surprises, ok?

```suggestion
FROM golang:1.23-alpine3.20
```

Reply:
good catch, thanks